site/src/pages/404.tsx

```diff
@@ -1,8 +1,64 @@
+import { Icon } from "@cultureamp/kaizen-component-library"
+import { Link } from "gatsby"
 import * as React from "react"
+import ContentMarkdownSection from "../components/ContentMarkdownSection"
+import { Content, ContentOnly } from "../components/ContentOnly"
+import Footer from "../components/Footer"
 import Layout from "../components/Layout"
+import md from "../components/markdownComponents"
+import PageHeader from "../components/PageHeader"
+
+const exclamationIcon = require("@cultureamp/kaizen-component-library/icons/exclamation.icon.svg")
+  .default
+
+const FourOhFourPageHeader = (
+  <PageHeader
+    headingText={
+      <span style={{ display: "flex", alignItems: "center" }}>
+        <span
+          style={{
+            display: "inline-block",
+            width: "1em",
+            marginRight: "0.3em",
+          }}
```

Comment:
I'd allow this in the very specific case of aligning icons to text, which is so dependent on font size and eyeballing (it doesn't generalise). But generally I think it's better to put it in a file, because it's less unexpected for the next person (I'd expect to see a scss source map to find the line of scss, not a tsx file, seeing as we use scss everywhere else). Also the specificity issue, but I can't imagine why anyone would ever want to override this very specific style.

For future consideration: manually aligning icons to text is such a common thing to need to do, that I wonder if the component itself could support it better, such as a prop for margin right and arbitrary size, so as to not need to wrap it in a span every time you need to do this.

Reply:
> I wonder if the component itself could support it better, such as a prop for margin right and arbitrary size

Which component do you mean? I was thinking it'd be great to have a more sensible automated way of doing this too, but this issue seems to be that there's no component that wraps both the icon and the text.

Reply:
I was think of a less automated way where you are just passing in margin right and size to the Icon component, to avoid having to wrap the icon in a span, but that might be too much of half-measure and wrapping in a span isn't so bad.

I wonder if it's even possible to fully automate aligning icons to text. It seems like there's always some level of aesthetic judgement for every font size and it depends on where it's used. Although there could be a sensible default. This has got me thinking about icon fonts... which suck (the consensus last time I checked 😆, I haven't looked into it enough to have a strong opinion ), but I guess they'd respond to font size, which is nice for this?

Reply:
Yeah maybe, I've always shied away from icon fonts so don't know much about them to be honest.

The margin thing on the icon could work but the real issue is the vertical alignment. I tend to wrap the icon + text in a flex container and do it that way. We could make a component that's essentially just a span with `display: flex; align-items: center` that has slots for an icon and text, but it just seems a bit restrictive/too simple for a component. I haven't yet thought of a good way around it. Part of me just wants to believe that aligning icons + text is just a CSS skill that people should learn, and the other part thinks the design system should make it really convenient. Argh.